src/Integration.Vsix/SLCore/SLCoreEmbeddedPluginJarLocator.cs

```diff
@@ -33,16 +34,24 @@ namespace SonarLint.VisualStudio.Integration.Vsix.SLCore;
 public class SLCoreEmbeddedPluginJarLocator : ISLCoreEmbeddedPluginJarLocator
 {
     private const string JarFolderName = "DownloadedJars";
+
+    private readonly Dictionary<string, string> connectedModePluginNamePatternByPluginKey = new()
+    {
+        { "text", "sonar-text-plugin-(\\d+\\.\\d+\\.\\d+\\.\\d+)\\.jar" },
```

Comment:
Probably would be simpler to understand if it was just `StartsWith` instead of regex. List jar files already returns only the jars and the version pattern is redundant at that point

Reply:
I think it is not redundant, but safer. If in some point we will have a sonar-text-plugin-enterprise-[*], it will match exactly the expected one. Therefore I will not change that